test/unit/StakingPool/processExpirations.js

```diff
@@ -108,4 +117,269 @@ describe('processExpirations', function () {
 
     await expect(stakingPool.processExpirations(true)).to.not.reverted;
   });
+
+  it('anyone can call this method', async function () {
+    const { stakingPool } = this;
+    const {
+      nonMembers: [anyone],
+    } = this.accounts;
+
+    await expect(stakingPool.connect(anyone).processExpirations(true)).to.not.be.reverted;
+  });
+
+  it('Expires tranches updating active stake, stake shares and rewards shares supply', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination, initialPoolFee } = processExpirationsFixture;
+
+    const { firstActiveTrancheId, maxTranche } = await getTranches();
+
+    const tranches = Array(maxTranche - firstActiveTrancheId + 1)
+      .fill(0)
+      .map((e, i) => firstActiveTrancheId + i);
+
+    let stakeSharesTotalSupply = parseEther('0');
+    for (let i = 0; i < tranches.length; i++) {
+      const tranche = tranches[i];
+      await stakingPool.connect(user).depositTo([
+        {
+          amount,
+          trancheId: tranche,
+          tokenId,
+          destination,
+        },
+      ]);
+      const nftId = i + 1;
+      const deposit = await stakingPool.deposits(nftId, tranche);
+
+      const feesRewardShares = deposit.rewardsShares.mul(initialPoolFee).div(POOL_FEE_DENOMINATOR);
+
+      stakeSharesTotalSupply = stakeSharesTotalSupply.add(deposit.rewardsShares.add(feesRewardShares));
+
+      const trancheData = await stakingPool.tranches(tranche);
+      expect(trancheData.stakeShares).to.equal(deposit.stakeShares);
+      expect(trancheData.rewardsShares).to.equal(deposit.rewardsShares.add(feesRewardShares));
+    }
+
+    const baseStakeShares = Math.sqrt(amount);
+    const depositsCount = 8;
+
+    {
+      const activeStake = await stakingPool.activeStake();
+      const stakeSharesSupply = await stakingPool.stakeSharesSupply();
+      const rewardsSharesSupply = await stakingPool.rewardsSharesSupply();
+
+      expect(activeStake).to.equal(amount.mul(depositsCount));
+      expect(stakeSharesSupply).to.equal(baseStakeShares * depositsCount);
+      expect(rewardsSharesSupply).to.equal(stakeSharesTotalSupply);
+    }
+
+    await generateRewards(stakingPool, this.coverSigner, TRANCHE_DURATION * 7, 0);
+
+    await stakingPool.processExpirations(true);
+
+    await increaseTime(TRANCHE_DURATION * 8);
+
+    // expire all tranches
+    await stakingPool.processExpirations(true);
+
+    // Validate tranches are expired
+    for (let i = 0; i < tranches.length; i++) {
+      const tranche = tranches[i];
+      const expiredTranche = await stakingPool.expiredTranches(tranche);
+
+      const activeDepositsAtTranche = maxTranche - tranche + 1;
+
+      expect(expiredTranche.accNxmPerRewardShareAtExpiry).to.gt(0);
```

Comment:
let's see if we can turn this into an to.be.equal

to strengthen the assertion

Reply:
I tried calculating the exact value but it was difficult to get it when all tranches expired at once without replicating the full contract logic from the `processExpirations()` method. That's why I left this assertion as it is and instead added a new test `'Expires tranches correctly storing expiredTranches struct` that expires one tranche at a time and validates the exact value for `expiredTranche.accNxmPerRewardShareAtExpiry` here https://github.com/NexusMutual/smart-contracts/pull/538/commits/328d5a5844af746ce1f0639ab9c679e855dc93f4

Reply:
Thanks @patitonar 

Sounds ok


@shark0der let me know your thoughts here as well

Reply:
Kinda torn on this one. How about implementing a test where we calculate the `accNxmPerRewardShare` when `processExpirations` expires `1 bucket + 1 tranche + 1 bucket` (in this order)? This way we can limit the number of calculations needed in the test itself but make sure that the calculation is done correctly.

Reply:
Great suggestion @shark0der ! I created a that new test `correctly calculates accNxmPerRewardShare` here [Add test for accNxmPerRewardShare calculation](https://github.com/NexusMutual/smart-contracts/pull/538/commits/6047d7b15751a5d71fba22daf9c3df2911229eb7)